File: BOJ/[BOJ 1138] 한 줄로 서기.py
# N = int(input())
# lst = list(map(int, input().split()))
# res = [0]*N

# for i in range(N):
#     count = 0
#     for j in range(N):
#         if res[j]==0:
#             if count == lst[i]:
#                 res[j]=i+1
#                 break
#             else:
#                 count+=1
# print(*res)


# 10분

import sys
input = sys.stdin.readline

N = int(input())
lst = list(map(int, input().split()))

result = [0]*N
for i in range(N):
  k = 0
  for j in range(N):
    if result[j] == 0:
      if k == lst[i]:
        result[j] = i+1
        break
      k += 1
print(*result)

File: BOJ/[BOJ 1138] 한 줄로 서기.js
const fs = require("fs");
const input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((i) => i.split(" ").map(Number));

const [[N], ...[lst]] = input;

const stack = Array.from({ length: N }, (_, i) => i);
const result = Array.from({ length: N }, () => 0);
let i = 1;
for (const num of lst) {
  const index = stack.splice(num, 1)[0];
  result[index] = i;
  i += 1;
}

console.log(result.join(" "));
